esrally/telemetry.py

```diff
@@ -1408,6 +1409,209 @@ def record(self):
+                    node_name, "ingest_pipeline_processor_count", ingest_pipeline_processor_count, meta_data=metadata
+                )
+
+                start_time = start_stats_processors[processor_name].get("stats", {}).get("time_in_millis", 0)
+                end_time = processor_stats.get("stats", {}).get("time_in_millis", 0)
+                ingest_pipeline_processor_time = end_time - start_time
+
+                self.metrics_store.put_value_node_level(
+                    node_name, "ingest_pipeline_processor_time", ingest_pipeline_processor_time, unit="ms", meta_data=metadata
+                )
+
+                start_failed = start_stats_processors[processor_name].get("stats", {}).get("failed", 0)
+                end_failed = processor_stats.get("stats", {}).get("failed", 0)
+                ingest_pipeline_processor_failed = end_failed - start_failed
+
+                self.metrics_store.put_value_node_level(
+                    node_name, "ingest_pipeline_processor_failed", ingest_pipeline_processor_failed, meta_data=metadata
+                )
+
+            # We have a top level pipeline stats obj, which contains the total time spent preprocessing documents in
+            # the ingest pipeline.
+            elif processor_name == "total":
+
+                metadata = {"pipeline_name": pipeline_name, "cluster_name": cluster_name}
+
+                start_count = start_stats_processors[processor_name].get("count", 0)
+                end_count = processor_stats.get("count", 0)
+                ingest_pipeline_pipeline_count = end_count - start_count
+
+                self.metrics_store.put_value_node_level(
+                    node_name, "ingest_pipeline_pipeline_count", ingest_pipeline_pipeline_count, meta_data=metadata
+                )
+
+                start_time = start_stats_processors[processor_name].get("time_in_millis", 0)
+                end_time = processor_stats.get("time_in_millis", 0)
+                ingest_pipeline_pipeline_time = end_time - start_time
+
+                self.metrics_store.put_value_node_level(
+                    node_name, "ingest_pipeline_pipeline_time", ingest_pipeline_pipeline_time, unit="ms", meta_data=metadata
+                )
+
+                start_failed = start_stats_processors[processor_name].get("failed", 0)
+                end_failed = processor_stats.get("failed", 0)
+                ingest_pipeline_pipeline_failed = end_failed - start_failed
+
+                self.metrics_store.put_value_node_level(
+                    node_name, "ingest_pipeline_pipeline_failed", ingest_pipeline_pipeline_failed, meta_data=metadata
+                )
+
+    def get_ingest_pipeline_stats(self):
+        # pylint: disable=import-outside-toplevel
+
+        import elasticsearch
+
+        summaries = {}
+        for cluster_name in self.specified_cluster_names:
+            try:
+                ingest_stats = self.clients[cluster_name].nodes.stats(metric="ingest")
```

Comment:
Just an FYI @sethmlarson - `ingest` is not documented as a valid param in the [Python client docs](https://elasticsearch-py.readthedocs.io/en/v7.14.0/api.html?highlight=node%20stats#elasticsearch.client.NodesClient.stats):
> metric – Limit the information returned to the specified metrics Valid choices: _all, breaker, fs, http, indices, jvm, os, process, thread_pool, transport, discovery, indexing_pressure

I can submit a PR to fix this, unless there's a specific reason it is undocumented.

Reply:
Thanks for reporting this, can you make the change here? https://github.com/elastic/elasticsearch/blob/master/rest-api-spec/src/main/resources/rest-api-spec/api/nodes.stats.json#L40